packages/story-editor/src/app/taxonomy/taxonomyProvider.js

```diff
@@ -145,50 +146,46 @@ function TaxonomyProvider(props) {
   );
 
   const addSearchResultsToCache = useCallback(
-    async (
-      taxonomy,
-      {
-        name,
-        // This is the per_page value Gutenberg is using
-        perPage = 20,
-      },
-      addNameToSelection = false
-    ) => {
+    async (taxonomy, args, addNameToSelection = false) => {
       let response = [];
       const termsEndpoint = taxonomy['_links']?.['wp:items']?.[0]?.href;
       if (!termsEndpoint) {
         return;
       }
       try {
-        response = await getTaxonomyTerm(termsEndpoint, {
-          search: name,
-          per_page: perPage,
-        });
+        response = await getTaxonomyTerm(termsEndpoint, args);
       } catch (e) {
-        // Do we wanna do anything here?
+        // TODO #9033
       }
 
       // Avoid update if we're not actually adding any terms here
       if (response.length < 1) {
-        return;
+        // eslint-disable-next-line consistent-return
+        return response;
       }
 
       // Format results to fit in our { [taxonomy]: { [slug]: term } } map
       const termResults = {
         [taxonomy.restBase]: dictionaryOnKey(response, 'slug'),
       };
+
       setTermCache((cache) => mergeNestedDictionaries(cache, termResults));
 
-      if (addNameToSelection) {
-        const selectedTermSlug = cleanForSlug(name);
+      if (addNameToSelection && args.search) {
+        const selectedTermSlug = cleanForSlug(args.search);
         const selectedTerm = response.find(
           (term) => term.slug === selectedTermSlug
         );
 
         if (selectedTerm) {
           addTermToSelection(taxonomy, selectedTerm);
         }
+      } else if (args.search) {
+        setFlatSearchResults(response);
       }
+
+      // eslint-disable-next-line consistent-return
+      return response;
```

Comment:
I wouldn't disable this, I still need to update my PR, but I'm thinking we can just do something like this on the areas where we're not returning anything:
```
console.error('some helpful issue message');
return [];
```

I guess I'll be adding this in my PR too tho, so I guess disabling it temporarily is fine so we're not doing the same work 🤷